File: 2680. Maximum OR/Solution.java
class Solution {
    public long maximumOr(int[] nums, int k) {
        if (nums.length == 1) return (nums[0] * 1L) << k;
        int high = 0;
        long res = 0;
        List<Long> p = new ArrayList<>();
        int[] bits = new int[32];
        for (int n : nums) {
            int i;
            for (i = 0; (1L << i) <= n; i++) if (((1L << i) & n) > 0) bits[i]++;
            if (i > high) {
                p = new ArrayList<>();
                p.add(1L * n);
                high = i;
            } else if (i == high) {
                p.add(1L * n);
            }
        }
        for (long q : p) {
            long r = q << k;
            for (int i = 0; i < 32; i++) {
                if (bits[i] == 0 || ((1L << i) & r) > 0) continue;
                if ((q & (1L << i)) > 0 && bits[i] == 1) continue;
                r |= (1L << i);
            }
            res = Math.max(res, r);
        }
        return res;
    }
}

File: 2680. Maximum OR/Solution.py
class Solution:
    def maximumOr(self, nums: List[int], k: int) -> int:
        if len(nums) == 1: return nums[0] << k
        high, res = 0, 0
        p = []
        bits = [0 for i in range(32)]
        for n in nums:
            i = 0
            while (1 << i) <= n:
                if ((1 << i) & n) > 0: bits[i] += 1
                i += 1
            if i > high:
                p = [n]
                high = i
            elif i == high:
                p.append(n)
        for q in p:
            r = q << k
            for i in range(32):
                if bits[i] == 0 or ((1 << i) & r) > 0: continue
                if ((1 << i) & q) > 0 and bits[i] == 1: continue
                r |= (1 << i)
            res = max(res, r)
        return res